### PR TITLE
[ODH-3094]: Implemented Docker Image Build

### PR DESCRIPTION
**Pull request by EgorAfutin**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 
-RUN pip install nose nose-exclude coverage \
- && mkdir -p /odh/python/.xunit-reports /odh/python/.coverage-reports
-RUN pip install pytz==2018.4
+
+RUN apt-get update -y \
+ && apt-get install maven -y \
+ && mvn dependency:get -DrepoUrl=http://central.maven.org/maven2/ -Dartifact=org.apache.spark:spark-streaming-kafka-0-10_2.11:2.3.0 \
+ && mvn dependency:get -DrepoUrl=http://central.maven.org/maven2/ -Dartifact=org.apache.spark:spark-sql-kafka-0-10_2.11:2.3.0 \
+ && cp $HOME/.m2/repository/org/apache/spark/spark-sql-kafka-0-10_2.11/2.3.0/spark-sql-kafka-0-10_2.11-2.3.0.jar /spark/jars/ \
```
Comment (needmorec) [inline Dockerfile]: you copy only 2 jar files, but we need all dependencies as well

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+ && cp $HOME/.m2/repository/org/apache/spark/spark-streaming-kafka-0-10_2.11/2.3.0/spark-streaming-kafka-0-10_2.11-2.3.0.jar /spark/jars/ \
+ && pip install nose nose-exclude coverage \
```
Comment (needmorec) [inline Dockerfile]: swap this lines with one below, please

File: settings.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<settings>
+    <mirrors>
+        <mirror>
+            <id>maven</id>
+            <url>http://151.101.84.215</url>
```
Comment (halfwhite) [inline settings.xml]: What is this ip? Can we avoid changing of settings.xml?

Comment (EgorAfutin) [reply settings.xml]: I've just tried to establish connection with maven using mirroring, but it doesn't not help.

File: Dockerfile [hunk 4]
```diff
@@ -1,8 +1,18 @@
...
+ADD dependencies.xml /dependencies/
+
+RUN apt-get update -y \
+ && apt-get install maven -y \
+ && mvn dependency:copy-dependencies -f /dependencies/dependencies.xml -DoutputDirectory=/spark/jars \
+    -Dhttps.proxyHost=$http_proxy_host -Dhttps.proxyPort=$http_proxy_port \
+ && pip install nose nose-exclude coverage \
+ && mkdir -p /odh/python/.xunit-reports /odh/python/.coverage-reports \
+ && pip install pytz==2018.4 \
+ && rm -r /dependencies
```
Comment (needmorec) [inline Dockerfile]: up this line please and place it below mvn dependency:copy

Comment (EgorAfutin) [reply Dockerfile]: done

File: Dockerfile [hunk 5]
```diff
@@ -1,8 +1,18 @@
...
+ADD dependencies.xml /dependencies/
+
+RUN apt-get update -y \
+ && apt-get install maven -y \
+ && mvn dependency:copy-dependencies -f /dependencies/dependencies.xml -DoutputDirectory=/spark/jars \
+    -Dhttps.proxyHost=$http_proxy_host -Dhttps.proxyPort=$http_proxy_port \
+ && rm -r /dependencies \
+ && pip install nose nose-exclude coverage \
+ && mkdir -p /odh/python/.xunit-reports /odh/python/.coverage-reports \
+ && pip install pytz==2018.4
```
Comment (needmorec) [inline Dockerfile]: move this up as well, let's keep all similar commands together

Comment (EgorAfutin) [reply Dockerfile]: done

**Pull request merged**